A4A > Tiers: Implement updated overview page

yashwin opened the pull request:
Related to https://linear.app/a8c/issue/A4A-1677/ui-implement-the-updated-agency-tier-page

Resolves https://linear.app/a8c/issue/A4A-1691/implement-influenced-revenue-section
Resolves https://linear.app/a8c/issue/A4A-1692/implement-tier-cards

## Proposed Changes

This PR implements the updated overview page for Agency Tiers. All the components here are designed to be able to move to the Multi-site Dashboard later with minimal changes.

## Why are these changes being made?

* To add the new overview page for Agency Tiers

## Testing Instructions

* Open the A4A live link
* Go to Agency Tier > Verify the UI looks as displayed below

<img width="1511" height="603" alt="Screenshot 2025-10-28 at 4 05 02 PM" src="https://github.com/user-attachments/assets/c2c4e004-11d9-45ec-8be1-037a73f33686" />

<img width="187" height="597" alt="Screenshot 2025-10-28 at 4 13 14 PM" src="https://github.com/user-attachments/assets/be571776-9c15-4bee-9757-77b1e7935026" />


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

yashwin committed:
A4A > Tiers: Implement Overview page

yashwin commented on client/assets/stylesheets/_wp-components-overrides.scss:
```diff
@@ -69,6 +69,15 @@
 }
 
+/* A4A */
+.theme-a8c-for-agencies {
```
~All the secondary buttons are different in color on A4A. This fixes it.~

yashwin commented on client/a8c-for-agencies/style.scss:
```diff
@@ -16,6 +16,7 @@
 	--color-text-white: var(--studio-white);
 	--color-text-black: var(--studio-black);
 	--wp-components-color-accent: var(--studio-automattic-blue-50);
+	--wp-admin-theme-color: var(--studio-automattic-blue-50);
```
There are some components like Stat that use this variable.

yashwin commented on client/a8c-for-agencies/sections/agency-tier/overview-content/constants.ts:
All the components here are designed to be able to move to the Multi-site Dashboard later with minimal changes.

jkguidaven commented on client/a8c-for-agencies/sections/agency-tier/overview-content/index.tsx:
```diff
@@ -0,0 +1,29 @@
+	currentAgencyTier?: AgencyTier;
+	totalInfluencedRevenue: number;
+} ) {
+	const isSmallViewport = useViewportMatch( 'huge', '<' );
```
Nitpicking: Should we consider using isLargeViewport here to maintain consistency with the breakpoint analogy?

yashwin replied:
Even I was thinking, but we want to handle the large screen after the huge breakpoints.

jkguidaven commented on client/a8c-for-agencies/sections/agency-tier/overview-content/constants.ts:
```diff
@@ -0,0 +1,60 @@
+	{
+		level: 0,
+		id: 'emerging-partner',
+		name: __( 'Account activated' ),
```
I'm curious if it is common practice in multi-dashboard to consider a translatable string constant. Technically, it is a dynamic value since it changes based on a variable.

yashwin replied:
Sorry, what do you mean by dynamic value here? It is all static names.

jkguidaven commented on client/a8c-for-agencies/sections/agency-tier/overview-content/influenced-revenue.tsx:
```diff
@@ -0,0 +1,34 @@
+	currentAgencyTier?: AgencyTier;
+	totalInfluencedRevenue: number;
+} ) {
+	const currentTier = ALL_TIERS.find( ( tier ) => tier.id === currentAgencyTier );
```
Perhaps we can use `currentAgencyTierId` as `currentAgencyTier` to avoid confusion about which is which. Then we can use `currentAgencyTier` as `currentTier`. wdyt?

jkguidaven commented on client/a8c-for-agencies/sections/agency-tier/primary/agency-tier-overview-v2/index.tsx:
```diff
@@ -24,7 +33,12 @@ export default function AgencyTierOverviewRevamped() {
 
-			<LayoutBody>Content</LayoutBody>
+			<LayoutBody>
+				<AgencyTierOverviewContent
```
I'm not sure when the agency Tier value is null, but if that happens, I assume we have different content for that case?

Would it make sense to optionally render this component here? This way, when we implement the component's content, we avoid assuming a potential null tier and simplify the rendering logic. wdyt?

yashwin replied:
I don't think we will have a null tier. But, I'll confirm this and address it separately if it exists.

jkguidaven reviewed (approved):
Works great! I have some minor comments which I don't think is blocker. Feel free to address them.

jkguidaven commented on client/a8c-for-agencies/sections/agency-tier/overview-content/types.ts:
```diff
@@ -0,0 +1,5 @@
+export type AgencyTier =
```
I suggest using `AgencyTierType` here to highlight that it is a type rather than an actual class with properties.

jkguidaven commented on client/a8c-for-agencies/sections/agency-tier/overview-content/tier-cards.tsx:
```diff
@@ -0,0 +1,83 @@
+	currentAgencyTier,
+	isSmallViewport,
+}: {
+	currentAgencyTier?: AgencyTier;
```
Related to my comment above, maybe we can rename this to currentAgencyTierId?

yashwin committed:
address PR comments

yashwin commented:
Thanks for the review, @jkguidaven!

I have addressed all your comments and made some changes: https://github.com/Automattic/wp-calypso/pull/106776/commits/f1d102e1a27970b410581d46751ccea5fb48d296

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/29277996

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @yashwin for including a screenshot in the description! This is really helpful for our translators.
